components/UnifiedDocFeed/utils/getSelectedUrlFilters.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  feedTypeOpts,
+  sortOpts,
+  tagFilters,
+  topLevelFilters,
+} from "../constants/UnifiedDocFilters";
+import { scopeOptions } from "~/config/utils/options";
+
+export type SelectedUrlFilters = {
+  topLevel: string | undefined;
+  type: string | undefined;
+  sort: string | undefined;
+  time: string | undefined;
+  tags: string[];
+};
+
+export const getSelectedUrlFilters = ({
+  query,
+  pathname,
+}): SelectedUrlFilters => {
+  const defaults = {
+    topLevel: topLevelFilters[0].value,
+    type: Object.values(feedTypeOpts)[0].value,
+    sort: sortOpts[0].value,
+    time: scopeOptions[0].value,
+    tags: <string[]>[],
+  };
+  const selected = { ...defaults };
+
+  if (!(query || pathname)) {
+    return defaults;
+  }
+
+  if (Array.isArray(query.tags)) {
+    selected.tags = [...query.tags];
+  } else if (query.tags) {
+    selected.tags.push(query.tags);
+  }
```

Comment:
obj mutation like this makes it hard to understand what's happening. Could we inline this at line 28 such that
```
const selected = { 
...defaults, 
tags: Array.isArray(query.tags) ? [...query.tags] : selected.tags.push(query.tags)
}
```